### ISSUE TITLE
GPU memory consumption drastically increased in onnxruntime-gpu==1.8.0

### ISSUE THREAD

**Issue by ivanst0** [microsoft/onnxruntime#8287]
**Describe the bug**
Evaluating an ONNX model consumed 130 MB of GPU memory in 1.7.0, while with 1.8.0 it takes 3.2 GB of GPU memory.

**Urgency**
Blocks the upgrade to 1.8.0.

**System information**
- OS Platform and Distribution: Windows
- ONNX Runtime installed from: onnxruntime-gpu Python package
- ONNX Runtime version: 1.8.0
- Python version: 3.6
- CUDA/cuDNN version: 11.2/8.x
- GPU model and memory: GeForce GTX 1070, 8 GB

**To Reproduce**
```
import numpy as np
import onnxruntime
import pynvml

pynvml.nvmlInit()
device_handle = pynvml.nvmlDeviceGetHandleByIndex(0)

def get_used_gpu_mem():
    return pynvml.nvmlDeviceGetMemoryInfo(device_handle).used

dnn_path = "model.onnx"
onnx_session = onnxruntime.InferenceSession(dnn_path)

input_name = onnx_session.get_inputs()[0].name
input_shape = onnx_session.get_inputs()[0].shape
input_img = np.random.rand(1, 3, 256, 256).astype(np.float32)

mem_before = get_used_gpu_mem()
onnx_session.run(None, {input_name: input_img})[0][0]
mem_after = get_used_gpu_mem()
print("before:   {:>15,}".format(mem_before))
print("after:    {:>15,}".format(mem_after))
print("increase: {:>15,}".format(mem_after - mem_before))
```

**with onnxruntime-gpu==1.7.0**:
```
before:     1,069,457,408
after:      1,199,480,832
increase:     130,023,424
```
**with onnxruntime-gpu==1.8.0**:
```
before:     1,071,685,632
after:      4,351,631,360
increase:   3,279,945,728
```

**Comment by yuslepukhin** [comment]
This is a known issue and will be addressed in 1.8.1

**Comment by hariharans29** [comment]
Fix shipped in 1.8.1
